hbase-client/src/main/java/org/apache/omid/transaction/TTable.java

```diff
@@ -228,21 +261,12 @@ public void delete(Transaction tx, Delete delete) throws IOException {
                 }
             }
         }
-        if (issueGet) {
-            // It's better to perform a transactional get to avoid deleting more
-            // than necessary
-            Result result = this.get(transaction, deleteG);
-            if (!result.isEmpty()) {
-                for (Entry<byte[], NavigableMap<byte[], NavigableMap<Long, byte[]>>> entryF : result.getMap()
-                    .entrySet()) {
-                    byte[] family = entryF.getKey();
-                    for (Entry<byte[], NavigableMap<Long, byte[]>> entryQ : entryF.getValue().entrySet()) {
-                        byte[] qualifier = entryQ.getKey();
-                        deleteP.add(family, qualifier, CellUtils.DELETE_TOMBSTONE);
-                        transaction.addWriteSetElement(new HBaseCellId(table, delete.getRow(), family, qualifier,
-                                                                       transaction.getStartTimestamp()));
-                    }
-                }
+
+        if (deleteFamily) {
```

Comment:
1. I guess you want to say this is optimized path for HBase backend? Just name the method to indicate this, then. 
2. I suspect there is a bug for non-HBase backends, because enforceHBaseTransactionManagerAsParam throws an exception if it does not succeed to downcast.